lib/active_job_status/job_tracker.rb

```diff
@@ -24,18 +25,28 @@ def performing
     end
 
     def completed
+      previous_status = store.fetch(job_id)
       store.write(
         job_id,
         JobStatus::COMPLETED.to_s
       )
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
     end
 
     def deleted
+      previous_status = store.fetch(job_id)
       store.delete(job_id)
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
     end
 
-    private
+    def remove_from_batch
```

Comment:
I think `remove_from_batch` should be a private method, it's only called by instance methods in this class.